sonic-chassisd/scripts/chassisd

```diff
@@ -630,6 +913,61 @@ class ChassisdDaemon(daemon_base.DaemonBase):
         else:
             self.log_warning("Caught unhandled signal '{}' - ignoring...".format(SIGNALS_TO_NAMES_DICT[sig]))
 
+    def set_initial_dpu_admin_state(self):
+        """Send admin_state trigger once to modules those are powered up"""
+        lock = threading.Lock()
+        threads = []
+        for module_index in range(0, self.module_updater.num_modules):
+            op = None
+            # Get midplane state of DPU
+            module_info_dict = self.module_updater._get_module_info(module_index)
+            module_name = platform_chassis.get_module(module_index).get_name()
+            state_db = daemon_base.db_connect("STATE_DB")
+            midplane_table = swsscommon.Table(state_db, CHASSIS_MIDPLANE_INFO_TABLE)
+            fvs = midplane_table.get(module_name)
+            midplane_state = 'False'
+            if isinstance(fvs, list) and fvs[0] is True:
+                fvs = dict(fvs[-1])
+                midplane_state = fvs[CHASSIS_MIDPLANE_INFO_ACCESS_FIELD]
+
+            try:
+                if module_info_dict is not None:
+                    # Get admin state of DPU
+                    key = module_info_dict[CHASSIS_MODULE_INFO_NAME_FIELD]
+                    admin_state = self.module_updater.get_module_admin_status(key)
+                    if admin_state == 'down' and midplane_state == 'True':
+                        # shutdown DPU
+                        op = MODULE_ADMIN_DOWN
+                    if admin_state != 'down' and midplane_state == 'False':
+                        # startup DPU
+                        op = MODULE_ADMIN_UP
+
+                    if op is not None:
+                        # Spawn a thread to handle the DPU logic
+                        def handle_dpu_logic():
+                            try:
+                                with lock:
+                                    module = self.module_updater.chassis.get_module(module_index)
```

Comment:
The `module_index` and `op` variables here are module specific, so these variables should be function arguments for `handle_dpu_logic`. With the current implementation there is a good probability that the `module_index` and `op` are already updated by the time a thread has started for the previous module, Example scenario:
`DPU1 : op = MODULE_ADMIN_UP and module_index = 1` thread.start() is called (but thread has still not started yet or lock is not yet obtained - > this is thread1)
We move to the next element in the for loop in `for module_index in range(0, self.module_updater.num_modules):`
`DPU2 : op = MODULE_ADMIN_DOWN and module_index = 2` thread.start() is called -> this is thread2.
At this point if thread1 starts executing the DPU2 related admin down operation is handled while the DPU1 admin up operation is skipped (Since `op` and `module_index` is already updated.